internal/cluster/cluster.go

```diff
@@ -1039,6 +1040,88 @@ func (m *Manager) IsReadyForInstallation(c *common.Cluster) (bool, string) {
 	return true, ""
 }
 
+func (m *Manager) detectAndStoreCollidingIPsForCluster(cluster *common.Cluster, db *gorm.DB) error {
+	if db == nil {
+		db = m.db
+	}
+	// We want to calculate ip collisions only when in pre-install states since it is needed for pre-install validations
+	allowedStates := []string{
+		models.ClusterStatusPendingForInput,
+		models.ClusterStatusInsufficient,
+
+		models.ClusterStatusReady,
+	}
+	if !funk.ContainsString(allowedStates, swag.StringValue(cluster.Status)) {
+		return nil
+	}
+
+	hosts := cluster.Hosts
+	sort.Slice(hosts, func(i, j int) bool {
+		return hosts[i].ID.String() < hosts[j].ID.String()
+	})
+	ipCollisions := make(map[string][]string)
+	collidingIPSWithMacs := make(map[string][]string)
+	for _, host := range hosts {
+		if len(host.Connectivity) > 0 {
+			connectivityReport, err := hostutil.UnmarshalConnectivityReport(host.Connectivity)
+			if err != nil {
+				// Let's not stop iterating over hosts but let's log this error.
+				m.log.WithError(err).Errorf("unable to unmarshall connectivity report for host %d due to error: %s", host.ID, err.Error())
+			} else {
+				collidingIPSWithMacs = getCollidingIPs(connectivityReport)
+				for k := range collidingIPSWithMacs {
+					ipCollisions[k] = collidingIPSWithMacs[k]
```

Comment:
Many hosts are iterated.  Why can a result from current host overwrite result from previous host? Shouldn't results be appended ?

Reply:
Please document the structure of the collision map with an example and what information we expect to retrieve from it.